lib/ontologies_api_client/request_federation.rb

```diff
@@ -20,7 +20,18 @@ def federated_get(params = {}, &link)
             end
           end
 
-          connections.flatten
+          connections = params[:is_collection] ? merge_collections(connections) : connections.flatten
+
+          connections
+        end
+
+        def merge_collections(connections)
```

Comment:
this can be generalized to the case of paginated responses, but for now as we have it only the search call, you can put this function only there.